### DISCUSSION TITLE
Deleting rows when only specific column has NA/missing value

### DISCUSSION THREAD

**Discussion by ANeeK181** [hosseinmoein/DataFrame#232] [General]
Hi

I have seen docs of drop_missing() and it only drops rows as per drop policy when all or some columns have NA values. I want to drop all rows when certain column (output) has missing values. How to proceed?

**Comment by hosseinmoein** [comment]
I am not sure I understand. can you show an example?

**Comment by ANeeK181** [comment]
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/aneekanwar/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/aneekanwar/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



A | B | C | D | E | OUTPUT
-- | -- | -- | -- | -- | --
1 | 1 | 1 | 1 | 1 | 1
2 | 2 | 2 | 2 | 2 | 2
3 | 3 | 3 | 3 | 3 |  
4 |   | 4 |   | 4 | 4
5 | 5 | 5 | 5 | 5 | 5
6 | 6 | 6 | 6 | 6 | 6
7 | 7 | 7 | 7 | 7 |  
8 |   | 8 | 8 | 8 | 8
9 | 9 | 9 | 9 | 9 | 9
10 | 10 | 10 |   | 10 | 10
11 | 11 | 11 | 11 | 11 | 11
12 | 12 | 12 | 12 | 12 | 12



</body>

</html>

**Comment by ANeeK181** [comment]
Imagine dataframe like above; I only want to delete rows where values are missing in OUTPUT column, but dont care about any other columns. So, i just want to drop rows no 3 and 7 since they dont have output label. So if I use drop_policy as any, it will also delete rows 4, 8 and 10 which is not what i want. Hope i am making some sense here

**Comment by hosseinmoein** [comment]
I see. you need a column-specific drop logic. I do not have that interface currently. You have to do it manually.
But I will put it on my todo list

**Comment by hosseinmoein** [comment]
Actually I take that back. You can use `remove_data_by_sel()`

**Comment by ANeeK181** [comment]
right, i see. I can use std::isnan to check for nan values in functor, and then remove rows contains nan in output column?

**Comment by ANeeK181** [comment]
still its a good feature to have in drop_missing() if you add it some day. Drop only those rows when some specific column has missing values
